src/util/taproot.rs

```diff
@@ -757,20 +761,68 @@ impl ControlBlock {
     }
 }
 
+/// Inner type representing future (non-tapscript) leaf versions. See [`LeafVersion::Future`].
+///
+/// NB: NO PUBLIC CONSTRUCTOR!
+/// The only way to construct this is by converting u8 to LeafVersion and then extracting it.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub struct FutureLeafVersion(u8);
+
+impl FutureLeafVersion {
+    pub(self) fn from_consensus(version: u8) -> Result<FutureLeafVersion, TaprootError> {
+        match version {
+            TAPROOT_LEAF_TAPSCRIPT => unreachable!("FutureLeafVersion::from_consensus should be never called for 0xC0 value"),
+            TAPROOT_ANNEX_PREFIX => Err(TaprootError::InvalidTaprootLeafVersion(TAPROOT_ANNEX_PREFIX)),
+            // The values that comply to this rule are the 32 even values between 0xc0 and 0xfe
+            0xC2 | 0xC4 | 0xC6 | 0xC8 | 0xCA | 0xCC | 0xCE |
+            0xD0 | 0xD2 | 0xD4 | 0xD6 | 0xD8 | 0xDA | 0xDC | 0xDE |
+            0xE0 | 0xE2 | 0xE4 | 0xE6 | 0xE8 | 0xEA | 0xEC | 0xEE |
+            0xF0 | 0xF2 | 0xF4 | 0xF6 | 0xF8 | 0xFA | 0xFC | 0xFE |
+            // and also 0x66, 0x7e, 0x80, 0x84, 0x96, 0x98, 0xba, 0xbc, 0xbe
+            0x66 | 0x7E | 0x80 | 0x84 | 0x96 | 0x98 | 0xBA | 0xBC | 0xBE => Ok(FutureLeafVersion(version)),
+            wrong => Err(TaprootError::InvalidTaprootLeafVersion(wrong))
+        }
+    }
+
+    /// Get consensus representation of the future leaf version.
+    #[inline]
+    pub fn into_consensus(self) -> u8 {
+        self.0
+    }
+}
+
+impl Into<u8> for FutureLeafVersion {
+    #[inline]
+    fn into(self) -> u8 {
+        self.into_consensus()
+    }
+}
+
+impl fmt::Display for FutureLeafVersion {
```

Comment:
@Kixunil I think this (unlike `LeafVersion`) should be allowed to be displayed as a simple number with whatever formatting options were used by the caller.

This is an internal type which can be obtained only from `match`ing over `LeafVersion` and thus will be rarely print directly.

Reply:
Sure, I don't feel strongly about the representation. But maybe also implement hex traits?

Reply:
TODO: create good first issue for hex traits

Reply:
I clearly remember that I was writing one here. Probably commit got lost through multiple rebases...